src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const separator = toFormat[3];
+
+  const dateArr = date.split(oldSeparator);
+
+  let yearIndex;
+
+  if (fromFormat.indexOf('YYYY') !== -1) {
+    yearIndex = fromFormat.indexOf('YYYY');
+  } else {
+    yearIndex = fromFormat.indexOf('YY');
```

Comment:
The code assumes that 'YYYY' or 'YY' will always be present in the `fromFormat`. It's important to handle cases where neither might be present, which could lead to an undefined `yearIndex`. Consider adding a check to ensure that a valid year index is found.